File: lecture007/wave1.java
import java.util.Scanner;
public class wave1{
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args){
        System.out.print("Enter row and col value : ");
        int row=scn.nextInt();
        int col=scn.nextInt();
        int [][] arr= new int[row][col];
        System.out.println("Enter array ");
        input(arr);
        System.out.println("new array is ");
        wave(arr);
    }

    public static void input(int[][] arr){
        for(int i=0 ; i<arr.length ; i++){
            for(int j=0 ; j<arr.length ; j++){
                arr[i][j]=scn.nextInt();
            }
        }
    }

    public static void wave(int [][] arr){
        int m=arr.length;
        int n=arr[0].length;
        for(int i =0 ; i<m ; i++){
            if(i%2==0){
                for(int j=0 ; j<n ; j++){
                    System.out.print(arr[i][j] + " ");
                }
            }
            else{
                for(int j=n-1 ; j>=0 ; j--){
                    System.out.print(arr[i][j] + " ");
                }
            }
            }
        }
    }

File: lecture007/wave1.cpp
#include<iostream>
#include<vector>
using namespace std;
void wave(vector<vector<int>> &arr){
    int m,n;
    n=arr.size();
    m=arr[0].size();
    for(int row=0; row<n ; row++){
        if(row%2==0){
            for(int col=0 ; col<m ; col++){
                cout<<arr[row][col]<<" ";
            }
        }
        else{
            for(int col=m-1 ; col>=0 ; col--){
                cout<<arr[row][col]<<" ";
            }
        }
    }
}
void input(vector<vector<int>> &arr){
    for(int row=0 ; row<arr.size() ; row++){
        for(int col=0 ; col< arr[0].size() ; col++){
            cin>>arr[row][col];
        }
    }
}
void display(vector<vector<int>> &arr){
    for(int row=0 ; row<arr.size() ; row++){
        for(int col=0 ; col< arr[0].size() ; col++){
            cout<<arr[row][col]<<" ";
        }
    }
}

int main(int args , char**argv){
    int row=0 , col=0;
    cin>>row>>col;
    vector<vector<int>> arr(row , vector<int>(col,0));
    input(arr);
    cout<<endl;
    wave(arr);
    //display(arr);
}